client/dashboard/profile-wizard/steps/theme/index.js

```diff
@@ -68,12 +70,56 @@ class Theme extends Component {
 		}
 	}
 
-	async onChoose( theme, location = '' ) {
+	onChoose( theme, location = '' ) {
 		const { updateProfileItems } = this.props;
+		const { price, slug } = theme;
+		const { activeTheme = '' } = getSetting( 'onboarding', {} );
+
+		this.setState( { chosen: slug } );
+		recordEvent( 'storeprofiler_store_theme_choose', { theme: slug, location } );
+
+		if ( theme !== activeTheme && getPriceValue( price ) <= 0 ) {
+			this.installTheme( slug );
+		} else {
+			updateProfileItems( { theme: slug } );
+		}
+	}
+
+	installTheme( slug ) {
+		const { createNotice } = this.props;
 
-		this.setState( { chosen: theme } );
-		recordEvent( 'storeprofiler_store_theme_choose', { theme, location } );
-		updateProfileItems( { theme } );
+		apiFetch( { path: '/wc-admin/onboarding/themes/install?theme=' + slug, method: 'POST' } )
+			.then( () => {
+				this.activateTheme( slug );
+			} )
+			.catch( response => {
+				this.setState( { chosen: null } );
+				createNotice( 'error', response.message );
```

Comment:
We don't need to do it now but it might be nice to bump a track here too to see how often this might fail.

Reply:
Good idea.  May add this to the server-side installation method so we can catch any REST requests that may end up utilizing this in the future.